chia/full_node/mempool_check_conditions.py

```diff
@@ -132,40 +134,222 @@ def mempool_assert_my_amount(condition: ConditionWithArgs, unspent: CoinRecord)
     return None
 
 
-def get_name_puzzle_conditions(generator: BlockGenerator, max_cost: int, safe_mode: bool) -> NPCResult:
+def parse_aggsig(args: SExp) -> List[bytes]:
+    pubkey = args.first().atom
+    args = args.rest()
+    message = args.first().atom
+    if len(pubkey) != 48:
+        raise ValidationError(Err.INVALID_CONDITION)
+    if len(message) > 1024:
+        raise ValidationError(Err.INVALID_CONDITION)
+    return [pubkey, message]
```

Comment:
all these `parse_*()` functions validates the input to some degree, and always truncates any additional garbage at the end of the argument list. The truncation is important partly to not carry around unnecessary heap allocations, but also allows for deduplication of conditions that are identical.

Reply:
maybe consts for MESSAGE_LENGTH and PUB_KEY would be nicer